core-bundle/src/Resources/contao/pages/PageError401.php

```diff
@@ -121,9 +122,12 @@ protected function prepare($objRootPage=null)
 			}
 
 			// Add the referer so the login module can redirect back
-			$referer = base64_encode(Environment::get('request'));
+			$url = $objNextPage->getAbsoluteUrl() . '?redirect=' . base64_encode(Environment::get('base') . Environment::get('request'));
```

Comment:
So far, the referer was only the path relative to the current base URL. Is it necessary to use absolute URLs?

Reply:
Yes, we either have to work with relative or absolute URLs. Checking the URI signed hash with a relative and an absolute URL will not equal to the same hash of course. But our redirect endpoints are generic so all of them have to work the same and we need cross-domain links e.g. for the FE preview links from the back end (if you have a multi domain setup).

Reply:
I don't think there will be cross-domain redirect requests. But it does not matter, and using absolute urls is consistent here. We might want the RedirectController to only accept relative to it's own domain, so the url to the redirect controller would need to be to the correct domain in the first place. But I think that's out of the scope of this PR, and #1118 might change stuff anyway 😉

Reply:
Ok then.